### ISSUE TITLE
Duplicate data attributes of input

### ISSUE THREAD

**Issue by aasensiog** [twitter/typeahead.js#1077]
I'm having problems with the component due to the hidden input created has the data-attributes of the original input copied.
I mean, having:
```html
<input type='text' id='x' name='x' data-role="x" class='form-control'/>
```
So I configure the typeahead for the object $('#x').
Once created I find:
```html
<input type='text' id='x' name='x' data-role="x" class='form-control tt-input' />
<input type='text' data-role="x" class='form-control tt-hint'/>
```
The problem is that, after I use $('[data-role=x]').val() and I expect only 1 result but I got 2 instead.

I would like that the created input for the component doesn't copy the data- attributes as doesn't copy name or id.
Thanks a lot.

**Comment by jharding** [comment]
Yeah... seems like copying the data attributes probably isn't a good idea.

**Comment by jharding** [comment]
Eh, been thinking about this more and I'm not sure what "correct" behavior should be. I'm inclined to leave it as is seeing as how I haven't heard any other reports of similar problems.

**Comment by jochenberger** [comment]
This also causes issues for the Apache Tapestry project that controls form validation via data attributes: https://issues.apache.org/jira/browse/TAP5-2440.

**Comment by GeorgeCrecoukias** [comment]
> It seems removeData() is not working correctly

It's working correctly I think. .removeData will only remove data values tracked internally on the element but won't remove data- attributes.

We have an issue with this as well, where inputs without the [name] attribute will conflict with each other if you are using jquery validator. So I'm currently looking for a way to remove the validation attributes from the hint input or even better have a name for it that won't conflict with other inputs in the DOM.

Update: We did this for the time being
 $("form").data("validator").settings.ignore = ":not([name])";
In document.ready event.

**Comment by erkurita** [comment]
The biggest issue with this problem is React.js. Since React assigns `data-react-id` attributes to most elements (if not all), when the typeahead is initialized it creates 2 inputs with the same `data-react-id` and React goes bonkers.

Perhaps we could have a setting with which turn on or off this behaviour?

**Comment by benedict-w** [comment]
I'm also getting Google Web Console crawler errors for duplicate itemprop="query-input" field on site search box
